src/logic/render/PointRenderEngine.ts

```diff
@@ -12,83 +12,70 @@ import {
     updateImageDataById
 } from "../../store/editor/actionCreators";
 import {RectUtil} from "../../utils/RectUtil";
-import _ from "lodash";
 import {DrawUtil} from "../../utils/DrawUtil";
 import {PointUtil} from "../../utils/PointUtil";
 import {updateCustomcursorStyle} from "../../store/general/actionCreators";
 import {CustomCursorStyle} from "../../data/CustomCursorStyle";
-import {BaseSuportRenderEngine} from "./BaseSuportRenderEngine";
-import {NumberUtil} from "../../utils/NumberUtil";
+import {BaseSupportRenderEngine} from "./BaseSupportRenderEngine";
+import {EditorSelector} from "../../store/selectors/EditorSelector";
+import {EditorData} from "../../data/EditorData";
 
-export class PointRenderEngine extends BaseSuportRenderEngine {
+export class PointRenderEngine extends BaseSupportRenderEngine {
     private config: RenderEngineConfig = new RenderEngineConfig();
 
     // =================================================================================================================
     // STATE
     // =================================================================================================================
 
     private transformInProgress: boolean;
-    private mousePosition: IPoint;
-    private scale: number;
 
-    public constructor(canvas: HTMLCanvasElement, imageRect: IRect) {
-        super(canvas, imageRect);
+    public constructor(canvas: HTMLCanvasElement) {
+        super(canvas);
     }
 
     // =================================================================================================================
     // EVENT HANDLERS
     // =================================================================================================================
 
-    public mouseDownHandler(event: MouseEvent): void {
-        const mousePosition: IPoint = CanvasUtil.getMousePositionOnCanvasFromEvent(event, this.canvas);
-        const isMouseOverImage: boolean = RectUtil.isPointInside(this.imageRectOnCanvas, mousePosition);
-        const isMouseOverCanvas: boolean = RectUtil.isPointInside({x: 0, y: 0, ...CanvasUtil.getSize(this.canvas)},
-            this.mousePosition);
+    public mouseDownHandler(data: EditorData): void {
```

Comment:
nie będę za bardzo wchodził w logikę tych metod, ale ogólnie rzecz biorąc wygląda to dobrze

Reply:
no i przede wszystkim czytelność się poprawiła